### PR TITLE
cna annotation +germline/somtic labeling

### PR DESCRIPTION
**Pull request by DouglasAbrams**

### CODE REVIEW

File: wgs/cohort_qc.py [hunk 1]
```diff
@@ -16,15 +16,43 @@ def cohort_qc_workflow(args):
...
+    cna_table = {label[0]: os.path.join(out_dir, label[0], "cna_table.tsv") for label, data in inputs.items()}
+    cohort_maf = {label[0]: os.path.join(out_dir, label[0], "cohort.maf") for label, data in inputs.items()}
+    cohort_maf_oncogenic_filtered = {label[0]: os.path.join(out_dir, label[0], "cohort_oncogenic_filtered.maf") for label, data in inputs.items()}
+    cohort_cna = {label[0]: os.path.join(out_dir, label[0], "cohort_cna.tsv") for label, data in inputs.items()}
 
     workflow.setobj(
         obj=mgd.OutputChunks('cohort_label', 'sample_label'),
         value=list(inputs.keys()),
     )
+    # workflow.subworkflow(
```
Comment (diljotgrewal) [inline wgs/cohort_qc.py]: remove please

File: wgs/cohort_qc.py [hunk 2]
```diff
@@ -35,9 +63,14 @@ def cohort_qc_workflow(args):
             api_key,
             out_dir,
             sample_labels, 
-            mgd.InputFile('sample_maf_dict', 'cohort_label', 'sample_label', fnames=sample_mafs, axes_origin=[]),
+            mgd.InputFile('germline_mafs_dict', 'cohort_label', 'sample_label', fnames=germline_mafs, axes_origin=[]),
+            mgd.InputFile('somatic_mafs_dict', 'cohort_label', 'sample_label', fnames=somatic_mafs, axes_origin=[]),
+            mgd.InputFile('cna_table', 'cohort_label', fnames=cna_table),
             mgd.OutputFile('report_path', 'cohort_label', fnames=report_path),
-        
+            # mgd.OutputFile('cohort_maf', 'cohort_label', fnames=cohort_maf),
```
Comment (diljotgrewal) [inline wgs/cohort_qc.py]: remove

File: wgs/cohort_qc.py [hunk 3]
```diff
@@ -16,15 +16,43 @@ def cohort_qc_workflow(args):
...
+    #     args=(
+    #         mgd.InputFile('remixt_dict', 'cohort_label', 'sample_label', fnames=remixt_data, axes_origin=[]),
+    #         # mgd.InputFile('remixt', 'cohort_label', 'sample_label', fnames=remixt_data),
+    #         mgd.OutputFile('cohort_cna', 'cohort_label', fnames=cna_table),
+    #     )
+    # )
+
+    workflow.subworkflow(
+        name="classifycopynumber",
+        func="wgs.workflows.cohort_qc.cna_annotaion_workflow",
```
Comment (diljotgrewal) [inline wgs/cohort_qc.py]: misspelled annotation

File: wgs/workflows/cohort_qc/__init__.py [hunk 4]
```diff
@@ -6,9 +6,49 @@
 from wgs.config import config
 
 
+def cna_annotaion_workflow(cohort, labels, remixt_dict, output_table):
```
Comment (diljotgrewal) [inline wgs/workflows/cohort_qc/__init__.py]: misspelled

File: wgs/workflows/cohort_qc/__init__.py [hunk 5]
```diff
@@ -6,9 +6,49 @@
...
+    workflow = pypeliner.workflow.Workflow(
+        ctx={'docker_image': config.containers('wgs')}
+    )
+    workflow.setobj(
+        obj=mgd.OutputChunks('sample_label'),
+        value=list(remixt_dict.keys()),
+    )
+    gtf="/home/mcphersa1/work/apolloanalysis/metadata/Homo_sapiens.GRCh37.73.gtf.gz"
```
Comment (diljotgrewal) [inline wgs/workflows/cohort_qc/__init__.py]: no hardcoding please

File: wgs/workflows/cohort_qc/__init__.py [hunk 6]
```diff
@@ -28,27 +68,49 @@ def create_cohort_qc_workflow(
 
 
     workflow.transform(
-        name='merge_sample_mafs',
-        func='wgs.workflows.cohort_qc.tasks.merge_mafs',
+        name='merge_germline_mafs',
+        func='wgs.workflows.cohort_qc.tasks.merge_relable_mafs',
```
Comment (diljotgrewal) [inline wgs/workflows/cohort_qc/__init__.py]: relable? misspelled?

File: wgs/workflows/cohort_qc/scripts/maftools_plots.R [hunk 7]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env Rscript
+
+library(maftools)
+
+
+# laml.plus.cn = read.maf(maf = "/juno/work/shah/abramsd/RESULTS/dlp_cohort_oncoplot/DATA/MAFS_COMPLETE/sample_mafs_merged_SA906_filt2", cnTable = "/juno/work/shah/abramsd/RESULTS/dlp_cohort_oncoplot/DATA/copynumber_classification/signatures/cna_annotations_rerun.tsv")
```
Comment (diljotgrewal) [inline wgs/workflows/cohort_qc/scripts/maftools_plots.R]: remove these comments

File: wgs/workflows/cohort_qc/tasks.py [hunk 8]
```diff
@@ -4,23 +4,75 @@
...
+
+def merge_relable_mafs(mafs, merged_maf, labels, class_label=None, write_header=True):
 
     mafs = list(mafs.values())
     labels = list(labels.values())
 
     #only write the first header
-    write_header = True
-
+    write_header = write_header
```
Comment (diljotgrewal) [inline wgs/workflows/cohort_qc/tasks.py]: what??

File: wgs/workflows/cohort_qc/tasks.py [hunk 9]
```diff
@@ -31,18 +83,24 @@ def annotate_maf_with_oncokb(
     cmd = [
         "MafAnnotator.py", "-i", maf, "-o", annotated_maf, "-b", api_key
     ]
+    cmd = [
+        "/juno/work/shah/abramsd/oncokb-annotator/MafAnnotator.py", "-i", maf, "-o", annotated_maf, "-b", api_key
```
Comment (diljotgrewal) [inline wgs/workflows/cohort_qc/tasks.py]: no hardcoding.

File: wgs/workflows/cohort_qc/tasks.py [hunk 10]
```diff
@@ -31,18 +83,24 @@ def annotate_maf_with_oncokb(
...
+    ]
 
     pypeliner.commandline.execute(*cmd, docker_image=docker_image)
 
 
 def filter_annotated_maf(annotated_maf, filtered_maf):
     annotated_maf = pd.read_csv(annotated_maf, sep="\t")
 
+    # filt_maf = annotated_maf[
```
Comment (diljotgrewal) [inline wgs/workflows/cohort_qc/tasks.py]: remove

File: wgs/workflows/cohort_qc/tasks.py [hunk 11]
```diff
@@ -80,3 +138,9 @@ def make_report(cohort_label, oncoplot, somatic_interactions, mafsummary, burden
         somatic_interactions, mafsummary, burden_plot
     ]
     pypeliner.commandline.execute(*cmd)
+
+
+c("Frame_Shift_Del_somatic", "Frame_Shift_Ins_somatic", "Splice_Site_somatic", "Translation_Start_Site_somatic",
```
Comment (diljotgrewal) [inline wgs/workflows/cohort_qc/tasks.py]: looks like R code pasted into python script??

File: wgs/workflows/cohort_qc/tasks.py (hunk 8 above)
Comment (DouglasAbrams) [reply wgs/workflows/cohort_qc/tasks.py]: write_header is an argument

File: wgs/workflows/cohort_qc/__init__.py (hunk 5 above)
Comment (DouglasAbrams) [reply wgs/workflows/cohort_qc/__init__.py]: can this be found in wgs refdir

File: wgs/workflows/cohort_qc/tasks.py (hunk 8 above)
Comment (diljotgrewal) [reply wgs/workflows/cohort_qc/tasks.py]: write_header = write_header

is redundant

File: wgs/workflows/cohort_qc/__init__.py (hunk 5 above)
Comment (diljotgrewal) [reply wgs/workflows/cohort_qc/__init__.py]: no, but we can add it.

File: wgs/workflows/cohort_qc/__init__.py [hunk 12]
```diff
@@ -6,9 +6,69 @@
...
 
+def cna_annotation_workflow(cohort, labels, remixt_dict, output_table, segmental_copynumber, gtf):
+    workflow = pypeliner.workflow.Workflow(
+        ctx={'docker_image': config.containers('wgs')}
+    )
+    workflow.setobj(
+        obj=mgd.OutputChunks('sample_label'),
+        value=list(remixt_dict.keys()),
+    )
+    # gtf="/home/mcphersa1/work/apolloanalysis/metadata/Homo_sapiens.GRCh37.73.gtf.gz"
```
Comment (diljotgrewal) [inline wgs/workflows/cohort_qc/__init__.py]: remove

File: wgs/workflows/cohort_qc/__init__.py [hunk 13]
```diff
@@ -57,22 +139,22 @@ def create_cohort_qc_workflow(
...
             args=(
-                mgd.TempInputFile("annotated_maf"),
-                mgd.OutputFile(filtered_maf),
+                mgd.InputFile(cohort_maf_oncogenic_annotated),
+                mgd.TempOutputFile("filtered_maf"),
             ),
         )
-
-        kwargs = {"filtered_maf": mgd.InputFile(filtered_maf)}
+        kwargs={"filtered_maf":  mgd.TempInputFile("filtered_maf"), 'docker_image':config.containers("wgs_qc_html") }
```
Comment (diljotgrewal) [inline wgs/workflows/cohort_qc/__init__.py]: break this long line up into 2

**Pull request merged**